sdk/servicebus/Azure.Messaging.ServiceBus/src/Plugins/ServiceBusPipeline.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using Azure.Messaging.ServiceBus.Diagnostics;
+using Azure.Messaging.ServiceBus.Plugins;
+
+namespace Azure.Messaging.ServiceBus
+{
+    internal class ServiceBusPipeline
+    {
+        private ReadOnlyMemory<ServiceBusPlugin> _pipeline;
+
+        public ServiceBusPipeline(ReadOnlyMemory<ServiceBusPlugin> pipeline)
+        {
+            _pipeline = pipeline;
+        }
+
+        public ValueTask ProcessSendAsync(ServiceBusMessage message)
+        {
+            if (_pipeline.Length > 0)
```

Comment:
Should we consider a method to handle the span/slice logic to make it easier for callers if we expose?   Maybe something like `MoveToNextStep` or similar?     Could be premature, but it would also remove the duplication in the two process methods.